### PR TITLE
Issue/3936 migrate products screens

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of #3936, this migrates the remaining products related screens to Hilt.

#### Testing
Smoke test the product's features.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductTypesBottomSheetFragment.kt [hunk 1]
```diff
@@ -17,22 +16,19 @@ import com.woocommerce.android.ui.products.ProductTypesBottomSheetViewModel.Prod
...
 
-class ProductTypesBottomSheetFragment : BottomSheetDialogFragment(), HasAndroidInjector {
+@AndroidEntryPoint
+class ProductTypesBottomSheetFragment : BottomSheetDialogFragment() {
     companion object {
         const val KEY_PRODUCT_TYPE_RESULT = "key_product_type_result"
     }
 
     @Inject internal lateinit var navigator: ProductNavigator
     @Inject internal lateinit var childInjector: DispatchingAndroidInjector<Any>
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductTypesBottomSheetFragment.kt]: Do we still need this?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSortingFragment.kt [hunk 2]
```diff
@@ -12,19 +11,16 @@ import com.google.android.material.bottomsheet.BottomSheetDialogFragment
...
-import dagger.android.HasAndroidInjector
-import dagger.android.support.AndroidSupportInjection
+import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
 
-class ProductSortingFragment : BottomSheetDialogFragment(), HasAndroidInjector {
+@AndroidEntryPoint
+class ProductSortingFragment : BottomSheetDialogFragment() {
     @Inject lateinit var currencyFormatter: CurrencyFormatter
     @Inject internal lateinit var childInjector: DispatchingAndroidInjector<Any>
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSortingFragment.kt]: Still needed?

**Review by kidinov — APPROVED** [review]
Looks good! Just a couple of comments

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSortingFragment.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSortingFragment.kt]: No, I keep forgetting them in the DialogFragments 🤦, I'll remove them.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSortingFragment.kt]: Done in 1240283

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductTypesBottomSheetFragment.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductTypesBottomSheetFragment.kt]: Fixed in 1240283

**Pull request merged**